user.py

```diff
@@ -157,6 +160,32 @@ def delete_researcher(user_id):
         db.session.commit()
 
 
+@user_bp.route('/<int:user_id>/preferences/<int:current_year>', methods=['GET'])
+@login_required
+@check_access_level(Role.ADMIN, Role.RESEARCHER)
+def preferences(user_id, current_year):
+    if not is_allowed_user(user_id):
```

Comment:
You should also check the user is a researcher. The page content has no sense otherwise, and moreover, the template rendering fails as `researcher` is None.